src/main/java/Duke.java

```diff
@@ -1,10 +1,127 @@
+     * Else echo back the message the user inputted
+     * @param message entered by user.
+     * @return boolean used to inform other function if user wants to quit
+     *     true -> continue using bot, false -> quit bot.
+     */
+    public boolean parseMessage(String message, TaskTracker tasks) {
+        try {
+            if (message.equals("bye")) {
+                printGoodBye();
+                return false;
+            } else if (message.equals("i")) {
+                botInstruction();
+                return true;
+            } else if (message.equals("list")) {
+                tasks.printStoredTask();
+                return true;
+            } else if (message.substring(0,4).equals("done")) {
+                tasks.doneTask(Integer.parseInt(message.split(" ",2)[1]));
+                return true;
+            } else if (message.substring(0,4).equals("todo")) {
+                tasks.addTask(message, Symbol.T);
+                return true;
+            } else if (message.substring(0,5).equals("event")) {
+                tasks.addTask(message, Symbol.E);
+                return true;
+            } else if (message.substring(0,8).equals("deadline")) {
+                tasks.addTask(message, Symbol.D);
+                return true;
+            } else if (message.substring(0,6).equals("delete")) {
+                tasks.deleteTask(message);
+                return true;
+            } else {
+                throw new IllegalArgumentException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
+        } catch (DukeException e) {
+            System.out.println(String.format("%80s",' ').replace(' ','*'));
+            System.out.println(String.format("* %-77s*",' '));
+            System.out.println(String.format("* %-77s*",e.getMessage()));
+            System.out.println(String.format("* %-77s*",' '));
+            System.out.println(String.format("%80s",' ').replace(' ','*'));
+            return true;
+        } catch (StringIndexOutOfBoundsException ex) {
+            System.out.println(String.format("%80s",' ').replace(' ','*'));
+            System.out.println(String.format("* %-77s*",' '));
+            System.out.println(String.format("* %-77s*","☹ OOPS!!! I'm sorry, but I don't know what that means :-("));
+            System.out.println(String.format("* %-77s*",' '));
+            System.out.println(String.format("%80s",' ').replace(' ','*'));
+            return true;
+        } catch (NullPointerException ex) {
+            System.out.println(String.format("%80s",' ').replace(' ','*'));
+            System.out.println(String.format("* %-77s*",' '));
+            System.out.println(String.format("* %-77s*",ex.getMessage()));
+            System.out.println(String.format("* %-77s*",' '));
+            System.out.println(String.format("%80s",' ').replace(' ','*'));
+            return true;
+        }
+    }
+
+    /**
+     * Echo message back in a text box.
```

Comment:
Change the word "Echo" to "Echoes" as starting verbs used for javadocs should be singular, present tense.